auth/http-client/snippets.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Demonstrates how to authenticate to Google Cloud Platform APIs using the
+Requests HTTP library."""
+
+import argparse
+
+
+def implicit():
+    import google.auth
+    from google.auth.transport import requests
```

Comment:
Why do we do these imports inside the function? And why do we have a vendored requests version?